docs/docker.md

```diff
@@ -0,0 +1,101 @@
+### Docker
+
+#### Building the image
+
+```
+docker build -t github/backup-utils .
+```
+
+#### Setting configuration options at runtime
+The `backup.config` file is dynamically populated at runtime with all `GHE_` environment variables that are part of the run command or Docker environment:
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-e "GHE_NUM_SNAPSHOTS=15" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+--rm \
+github/backup-utils ghe-backup
+```
+
+It is also possible to specify a `-e GHE_BACKUP_CONFIG` flag and volume mount in a local `backup.config` file rather than specify the variables individually at run time:
+
+```
+$ docker run -it  -e "GHE_BACKUP_CONFIG=/mnt/backup.config" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+-v $HOME/backup-utils/backup.config:/mnt/backup.config \
+--rm \
+github/backup-utils ghe-backup
+```
+
+#### SSH Keys
+
+A SSH private key that has been added to the GitHub Enterprise [Management Console for administrative SSH access](https://help.github.com/enterprise/admin/guides/installation/administrative-shell-ssh-access/) needs to be mounted into the container from the host system. It is also recommended to mount a SSH `.ssh/known_hosts` file into the container.
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-v ghe-backup-data:/data \
```

Comment:
Should probably quote this to be safe as peeps are likely to copy & paste and then change inline.